clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/DeployOpenstackAtomicOperation.groovy

```diff
@@ -41,20 +45,48 @@ class DeployOpenstackAtomicOperation implements AtomicOperation<Void> {
   * curl -X GET -H "Accept: application/json" localhost:7002/task/1
   */
   @Override
-  Void operate(List priorOutputs) {
-    task.updateStatus BASE_PHASE, "Initializing creation of Heat stack"
+  DeploymentResult operate(List priorOutputs) {
+    DeploymentResult deploymentResult = new DeploymentResult()
+    try {
+      task.updateStatus BASE_PHASE, "Initializing creation of server group"
+      OpenstackClientProvider provider = description.credentials.provider
 
-    def serverGroupNameResolver = new OpenstackServerGroupNameResolver(description.credentials, description.region)
-    def groupName = serverGroupNameResolver.combineAppStackDetail(description.application, description.stack, description.freeFormDetails)
+      def serverGroupNameResolver = new OpenstackServerGroupNameResolver(description.credentials, description.region)
+      def groupName = serverGroupNameResolver.combineAppStackDetail(description.application, description.stack, description.freeFormDetails)
 
-    task.updateStatus BASE_PHASE, "Looking up next sequence index for cluster ${groupName}..."
-    def stackName = serverGroupNameResolver.resolveNextServerGroupName(description.application, description.stack, description.freeFormDetails, false)
-    task.updateStatus BASE_PHASE, "Heat stack name chosen to be ${stackName}."
+      task.updateStatus BASE_PHASE, "Looking up next sequence index for cluster ${groupName}..."
+      def stackName = serverGroupNameResolver.resolveNextServerGroupName(description.application, description.stack, description.freeFormDetails, false)
+      task.updateStatus BASE_PHASE, "Heat stack name chosen to be ${stackName}."
 
+      task.updateStatus BASE_PHASE, "Loading templates"
+      String templateFile = 'asg.yaml'
+      String template = IOUtils.toString(this.class.classLoader.getResourceAsStream(templateFile))
+      String subtemplateFile = 'asg_resource.yaml'
+      String subtemplate = IOUtils.toString(this.class.classLoader.getResourceAsStream(subtemplateFile))
```

Comment:
The gain is not repeating ourselves, lines 62 - 65 look like copy/paste. 

Thinking something like this if you don't want to modify IOUtils:
`
Map<String, String> templateMap = ['asg.yaml', 'asg_resource.yaml'].collectEntries {
   [(it): IOUtils.toString(this.class.classLoader.getResourceAsString(it))]
}
`